modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EChain.java

```diff
@@ -233,6 +233,7 @@ private void analyzeWrite(final CompilerSettings settings, final Definition defi
         if (last instanceof ADefLink) {
             final ADefLink lastDef = (ADefLink) last;
             expression.analyze(settings, definition, variables);
+            // TODO: does it make more sense to just re-use last.after instead of using storeValueType?
```

Comment:
"marker" interface is also fine. I made it as a class for the special cases we might introduce when we have more "def" types propagation. For now interface is also fine. It would spare to repeat the ctor!